qt/contentrating.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2017 Jan Grulich <jgrulich@redhat.com>
+ * Copyright (C) 2016 Matthias Klumpp <matthias@tenstral.net>
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "appstream.h"
+#include "contentrating.h"
+
+#include <QDebug>
+#include "chelpers.h"
+
+using namespace AppStream;
+
+class AppStream::ContentRatingData : public QSharedData {
+public:
+    ContentRatingData()
+    {
+        m_contentRating = as_content_rating_new();
+    }
+
+    ContentRatingData(AsContentRating* cat) : m_contentRating(cat)
+    {
+        g_object_ref(m_contentRating);
+    }
+
+    ~ContentRatingData()
+    {
+        g_object_unref(m_contentRating);
+    }
+
+    bool operator==(const ContentRatingData& rd) const
+    {
+        return rd.m_contentRating == m_contentRating;
+    }
+
+    AsContentRating *contentRating() const
+    {
+        return m_contentRating;
+    }
+
+    AsContentRating* m_contentRating;
+};
+
+typedef QHash<ContentRating::RatingValue, QString> RatingMap;
```

Comment:
Not sure if this would make any difference, QHash is used accross appstream same way so I used it here as well.